src/common/lib/client/realtimechannel.ts

```diff
@@ -621,6 +621,39 @@ class RealtimeChannel extends EventEmitter {
         break;
       }
 
+      case actions.STATE: {
+        if (!this._liveObjects) {
+          return;
+        }
+
+        const { id, connectionId, timestamp } = message;
+        const options = this.channelOptions;
+
+        const stateMessages = message.state ?? [];
```

Comment:
could go a step further and move this decoding step into the plugin as well? would help to reduce the size impact on the default bundle

Reply:
I have slightly different thing on my TODO list:
[presence ](https://github.com/ably/ably-js/blob/a9f916939a12752f478c21972a08dd732bf729ac/src/common/lib/client/realtimechannel.ts#L606-L608)and [regular ](https://github.com/ably/ably-js/blob/a9f916939a12752f478c21972a08dd732bf729ac/src/common/lib/client/realtimechannel.ts#L757-L759)messages processing have same logic in them, I want to extract common bits for all of them, so we don't have duplicate code in the default bundle and the plugin, and it should also decrease the default bundle size.

are you ok with me doing it in the separate PR?

Reply:
yep that sounds good to me, might be good to make an issue too just for visibility?

Reply:
@coderabbitai can you create an issue for this?

> I have slightly different thing on my TODO list:
[presence ](https://github.com/ably/ably-js/blob/a9f916939a12752f478c21972a08dd732bf729ac/src/common/lib/client/realtimechannel.ts#L606-L608)and [regular ](https://github.com/ably/ably-js/blob/a9f916939a12752f478c21972a08dd732bf729ac/src/common/lib/client/realtimechannel.ts#L757-L759)messages processing have same logic in them, I want to extract common bits for all of them, so we don't have duplicate code in the default bundle and the plugin, and it should also decrease the default bundle size.